extensions/src/main/java/org/pentaho/platform/plugin/services/email/EmailConfigurationXml.java

```diff
@@ -189,6 +207,15 @@ public static Document getDocument( final IEmailConfiguration emailConfiguration
     String rawPassword = ObjectUtils.toString( emailConfiguration.getPassword() );
     setValue( document, PASSWORD_XPATH,
       StringUtils.isEmpty( rawPassword ) ? "" : Encr.encryptPasswordIfNotUsingVariables( rawPassword ) );
+    setValue( document, AUTH_MECHANISM_XPATH, ObjectUtils.toString( emailConfiguration.getAuthMechanism() ) );
+    setValue( document, CLIENT_ID_XPATH, ObjectUtils.toString( emailConfiguration.getClientId() ) );
+    setValue( document, CLIENT_SECRET_XPATH, ObjectUtils.toString( emailConfiguration.getClientSecret() ) );
+    setValue( document, SCOPE_XPATH, ObjectUtils.toString( emailConfiguration.getScope() ) );
+    setValue( document, GRANT_TYPE_XPATH, ObjectUtils.toString( emailConfiguration.getGrantType() ) );
```

Comment:
Need to update EmailConfigurationXmlTest.java to take into account all the new variables.